noir-projects/aztec-nr/aztec/src/oracle/header.nr

```diff
@@ -12,16 +15,17 @@ unconstrained pub fn get_header_at_internal(block_number: u32) -> Header {
 }
 
 pub fn get_header_at(block_number: u32, context: PrivateContext) -> Header {
-    let historical_header_block_number = context.historical_header.global_variables.block_number as u32;
+    let historical_header = context.historical_header;
+    let historical_block_number = historical_header.global_variables.block_number as u32;
 
-    if (block_number == historical_header_block_number) {
+    if (block_number == historical_block_number) {
         // If the block number we want to prove against is the same as the block number in the historical header we
         // skip the inclusion proofs and just return the historical header from context.
-        context.historical_header
+        historical_header
     } else {
         // 1) Get block number corresponding to the last_archive root in the header
         // Note: We subtract 1 because the last_archive root is the root of the archive after applying the previous block
-        let last_archive_block_number = historical_header_block_number - 1;
+        let last_archive_block_number = historical_block_number - 1;
```

Comment:
Eh if the goal is to prevent an oracle failure might as well do `assert(block_number < current_block_number)` (which simply prevents future blocks due to the previous `if`), and leave the archive fetching internals up the constraining - it's a bit out of place as it has to do with how the constraning is done. But it's not ultimately very important, if anything this seems a symptom of proofs being tied to headers.